docs/transaction.rst

```diff
@@ -0,0 +1,225 @@
+    user1_statement = BankStatement('user1', account_balance=2000, is_active=True)
+    user2_statement = BankStatement('user2', account_balance=0, is_active=True)
+
+    user1_statement.save()
+    user2_statement.save()
+
+    connection = Connection()
+
+    with TransactWrite(connection=connection, client_request_token='super-unique-key') as transaction:
+        # attempting to transfer funds from user1's account to user2's
+        transfer_amount = 1000
+        transaction.update(
+            BankStatement(user_id='user1'),
+            actions=[BankStatement.account_balance.add(transfer_amount * -1)],
+            condition=(
+                (BankStatement.account_balance >= transfer_amount) &
+                (BankStatement.is_active == True)
+            )
+        )
+        transaction.update(
+            BankStatement(user_id='user2'),
+            actions=[BankStatement.account_balance.add(transfer_amount)],
+            condition=(BankStatement.is_active == True)
+        )
+
+    user1_statement.refresh()
+    user2_statement.refresh()
+
+    assert user1_statement.account_balance == 1000
+    assert user2_statement.account_balance == 1000
+
+
+Now, say you make another attempt to debit one of the accounts when they don't have enough money in the bank:
+
+.. code-block:: python
+
+    from pynamodb.exceptions import TransactWriteError
+
+    assert user1_statement.account_balance == 1000
+    assert user2_statement.account_balance == 1000
+
+    try:
+        with TransactWrite(connection=connection, client_request_token='another-super-unique-key') as transaction:
+            # attempting to transfer funds from user1's account to user2's
+            transfer_amount = 2000
+            transaction.update(
+                BankStatement(user_id='user1'),
+                actions=[BankStatement.account_balance.add(transfer_amount * -1)],
+                condition=(
+                    (BankStatement.account_balance >= transfer_amount) &
+                    (BankStatement.is_active == True)
+                )
+            )
+            transaction.update(
+                BankStatement(user_id='user2'),
+                actions=[BankStatement.account_balance.add(transfer_amount)],
+                condition=(BankStatement.is_active == True)
+            )
+    except TransactWriteError as e:
+        # Because the condition check on the account balance failed,
+        # the entire transaction should be cancelled
+        assert e.cause_response_code == 'TransactionCanceledException'
+
+        user1_statement.refresh()
+        user2_statement.refresh()
+        # and both models should be unchanged
+        assert user1_statement.account_balance == 1000
+        assert user2_statement.account_balance == 1000
+
+
+Condition Check
+---------------
+
+The `ConditionCheck`:code: operation is used on a `TransactWrite`:code: to check if the current state of a record you
```

Comment:
when referring to classes, you can use the following syntax to link directly to api docs for them:
```
:py:class:`UnicodeAttribute <pynamodb.attributes.UnicodeAttribute>`
```